faculty/cli/cli.py

```diff
@@ -0,0 +1,1193 @@
+"""Command line interface."""
+
+# Copyright 2016-2018 ASI Data Science
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import division
+
+import contextlib
+import operator
+import os
+import os.path
+import shutil
+import stat
+import subprocess
+import sys
+import tempfile
+import textwrap
+import time
+import uuid
+from distutils.version import StrictVersion
+
+import click
+import requests
+from tabulate import tabulate
+
+import faculty.cli.auth
+import faculty.cli.baskerville
+import faculty.cli.casebook
+import faculty.cli.client
+import faculty.cli.config
+import faculty.cli.galleon
+import faculty.cli.hound
+import faculty.cli.parse
+import faculty.cli.shell
+import faculty.cli.update
+import faculty.cli.version
+import faculty.config
+from faculty.clients.base import NotFound
+
+SSH_OPTIONS = [
+    "-o",
+    "IdentitiesOnly=yes",
+    "-o",
+    "StrictHostKeyChecking=no",
+    "-o",
+    "BatchMode=yes",
+]
+
+
+class AmbiguousNameError(Exception):
+    """Exception when name matches multiple servers."""
+
+    pass
+
+
+def _print_and_exit(msg, code):
+    """Print error message and exit with given code."""
+    click.echo(msg, err=True)
+    sys.exit(code)
+
+
+def _get_pypi_versions():
+    """List releases available from PyPI."""
+    # TODO needs to point to the new repo
+    response = requests.get("https://pypi.python.org/pypi/sml/json", timeout=1)
+    versions = response.json()["releases"].keys()
+    return [StrictVersion(v) for v in versions]
+
+
+def _populate_creds_file():
+    """Prompt user for client ID and secret and save them."""
+    while True:
+        domain = click.prompt(
+            "Domain", default="services.sherlockml.com", err=True
```

Comment:
Update this to `services.faculty.ai` (or whatever the services domain for cloud is going to be)?

Reply:
`services.cloud.my.faculty.ai` says @neb42.